src/creature.h

```diff
@@ -506,6 +511,13 @@ class Creature
 
  public:
         body_part select_body_part(Creature *source, int hit_roll) const;
+        long get_id() const {
+            return global_id;
+        }
+
+        void set_id( long new_id ) {
```

Comment:
Would be nice to have a check here that prevents overriding an existing id. Initialize the global id to -1 or 0 or whatever fixed value in the constructor (marking it as "invalid" / "not set") and check here that one can only assign an actual id if the current is is invalid.

This would prevent (or at least report) accidental overriding, see `player::setID`